godtools/App/Services/Renderer/Views/MobileContent/Views/Pages/MobileContentPagesViewModel.swift

```diff
@@ -281,54 +281,30 @@ extension MobileContentPagesViewModel {
         switch page {
             
         case .pageId(let value):
-                     
-            if isChooseYourOwnAdventure {
-                
-                let introPageId: String = "intro"
-                let categoriesPageId: String = "categories"
-                let pageIsCategoryPage: Bool = value != categoriesPageId && value != introPageId
-                
-                guard pageIsCategoryPage else {
-                    return
-                }
-                
-                guard let pageModelMatchingPageId = renderablePages.first(where: {$0.id == value}) else {
-                    return
-                }
-                
-                if let introPage = renderablePages.first(where: {$0.id == introPageId}),
-                   let categoriesPage = renderablePages.first(where: {$0.id == categoriesPageId}) {
-                    
-                    pagesToRender = [introPage, categoriesPage, pageModelMatchingPageId]
-                }
-                else {
-                    
-                    pagesToRender = [pageModelMatchingPageId]
-                }
-                
-                navigateToPageModel = pageModelMatchingPageId
-                shouldReloadPagesUI = true
+                   
+            guard let pageModelMatchingPageId = renderablePages.first(where: {$0.id == value}) else {
+                return
             }
-            else {
-                
-                var pageModelsToRenderUpToPageToNavigateTo: [Page] = Array()
-                var pageModelMatchingPageId: Page?
+            
+            var pageModelsToRenderUpToPageToNavigateTo: [Page] = [pageModelMatchingPageId]
+            
+            while true {
+            
+                guard let parentPage = pageModelsToRenderUpToPageToNavigateTo[0].parentPage else {
+                    break
+                }
                 
-                for pageModel in renderablePages {
-                    
-                    pageModelsToRenderUpToPageToNavigateTo.append(pageModel)
-                    
-                    if pageModel.id == value {
-                        pageModelMatchingPageId = pageModel
-                        break
-                    }
+                guard !pageModelsToRenderUpToPageToNavigateTo.contains(parentPage) else {
+                    break
                 }
                 
-                pagesToRender = pageModelsToRenderUpToPageToNavigateTo
-                navigateToPageModel = pageModelMatchingPageId
-                shouldReloadPagesUI = true
+                pageModelsToRenderUpToPageToNavigateTo.insert(parentPage, at: 0)
```

Comment:
because you are using the `insert` method here, can you use `let` instead of `var` for `pageModelsToRenderUpToPageToNavigateTo`?

Reply:
Swift doesn't allow the ```insert``` on ``immutable``` since ```Array``` is a value type.

<img width="1237" alt="Screenshot 2023-03-03 at 2 16 36 PM" src="https://user-images.githubusercontent.com/59846460/222808050-ca28c2f5-bdb8-4d2f-9dcb-57e8c5d61166.png">

Reply:
k, wasn't sure the semantics of it :)